### PR TITLE
Triggering builds with UI

### PR DESCRIPTION
**Pull request by jhellar**
# Motivation

https://issues.jboss.org/browse/AEROGEAR-7922

# Verification steps

1. deploy local OpenShift with mobile-core
2. provision mobile CI/CD service
3. create mobile client
4. open mobile client's details and navigate to builds tab
5. create new build configuration (you can use https://github.com/aerogear/cordova-showcase-template.git)
6. fetch and checkout this PR
7. `make serve`
8. open `localhost:4000` in your browser
9. open the mobile app and navigate to builds tab
10. expand build config
11. click on start build button
12. new build should start

### CODE REVIEW

File: ui/src/components/build/NoBuild.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import React, { Component } from 'react';
-import { Button } from 'patternfly-react';
+import StartBuildButton from '../../containers/StartBuildButton';
 
 import './NoBuild.css';
```
Comment (wojta) [inline ui/src/components/build/NoBuild.js]: We should probably convert non-state simple rendering components to the function-style. (as @josemigallas recommends).

**Comment by psturc** [comment]
@jhellar not sure if @wojta thought the same, but I meant something like:
```
const DeleteBuildButton = (props) => {
    triggerBuild = () => {
      props.triggerBuild(props.jobName);
    }
    return ( <Button onClick={this.triggerBuild}>Start Build</Button> );
}
```
on the other hand, I haven't found any good article saying that it is a good and recommended practice to use stateless components with redux... @wojta any ideas?

**Comment by secondsun** [comment]
@jhellar How do I test this?  I've merged the pr locally and deployed it to a cluster with the mobile ci/cd running

**Comment by wojta** [comment]
@psturc  My experience with React is limited, so I'm not sure. But ESLint said that to me.
image

**Comment by jhellar** [comment]
@secondsun I've added verification steps to the description

**Comment by jhellar** [comment]
@psturc regarding the `StartBuildButton` component - from quick googling ('react stateless component event handler') there are few options if we want to convert the component to be stateless. The two most suggested are:
- pass the handler as prop from parent component - this component does only one thing, so probably not good approach
- make the component stateful

There are also other options:
- have the event handler inside the component (similarly to example in your comment) - drawback is that every time component is rendered, new function is created (but IMO in our case that should not be a problem)
- have the event handler outside of the component

WDYT?

**Comment by secondsun** [comment]
@jhellar I get these errors 
```
export STATIC_FILES_DIR=/home/summerspittman/go/src/github.com/aerogear/mobile-client-service/ui/build; ./mobile-client-service -kubeconfig ~/.kube/config
time="2018-09-26T07:37:09-04:00" level=info msg="Starting application" listenAddress=":4000"
time="2018-09-26T07:37:09-04:00" level=info msg="invalid secret aerogear-digger"
time="2018-09-26T07:37:09-04:00" level=info msg="invalid secret android-sdk-dockercfg-t7btm"
time="2018-09-26T07:37:09-04:00" level=info msg="invalid secret android-sdk-token-2r7sv"
time="2018-09-26T07:37:09-04:00" level=info msg="invalid secret android-sdk-token-8j8cg"
time="2018-09-26T07:37:09-04:00" level=info msg="invalid secret builder-dockercfg-v6xgp"
time="2018-09-26T07:37:09-04:00" level=info msg="invalid secret builder-token-ktl6v"
time="2018-09-26T07:37:09-04:00" level=info msg="invalid secret builder-token-mvb9d"
time="2018-09-26T07:37:09-04:00" level=info msg="invalid secret default-dockercfg-2v5v4"
time="2018-09-26T07:37:09-04:00" level=info msg="invalid secret default-token-24fvq"
time="2018-09-26T07:37:09-04:00" level=info msg="invalid secret default-token-h9dgd"
time="2018-09-26T07:37:09-04:00" level=info msg="invalid secret deployer-dockercfg-btg69"
time="2018-09-26T07:37:09-04:00" level=info msg="invalid secret deployer-token-kd5kk"
time="2018-09-26T07:37:09-04:00" level=info msg="invalid secret deployer-token-n88vc"
time="2018-09-26T07:37:09-04:00" level=info msg="invalid secret jenkins-dockercfg-4lm4n"
time="2018-09-26T07:37:09-04:00" level=info msg="invalid secret jenkins-token-j928f"
time="2018-09-26T07:37:09-04:00" level=info msg="invalid secret jenkins-token-m7bph"
time="2018-09-26T07:37:09-04:00" level=info msg="invalid secret localregistry-aerogear-digger-apb-parameters8ohyz"
time="2018-09-26T07:37:09-04:00" level=info msg="invalid secret mdc-server-server"
time="2018-09-26T07:37:09-04:00" level=info msg="invalid secret mobile-developer-console-dockercfg-24xrn"
time="2018-09-26T07:37:09-04:00" level=info msg="invalid secret mobile-developer-console-token-fbxfc"
time="2018-09-26T07:37:09-04:00" level=info msg="invalid secret mobile-developer-console-token-m644d"

```

**Comment by jhellar** [comment]
@secondsun these are there already from the POC work. They are generated by the operator looking for a secrets to build `mobile-services.json`s. Probably something we can improve in future.

**Comment by secondsun** [comment]
I'm not sure how to create the build configuration.  The openshift mobile UI seems to be removed from core and the create build button doesn't do anything

**Comment by psturc** [comment]
Can we merge this one?

**Pull request merged**